app/lib/actions/katello/pulp_selector.rb

```diff
@@ -2,12 +2,31 @@ module Actions
   module Katello
     class PulpSelector < Actions::Base
       def plan(backend_actions, repository, smart_proxy, *args)
+        pass_only_args = args.dig(0, :args_only) || false
+        return_action_output = args.dig(0, :return_output) || false
         backend_type = smart_proxy.backend_service_type(repository)
         found_action = backend_actions.find { |action| action.backend_service_type == backend_type }
 
         fail "Could not locate an action for type #{backend_type}" unless found_action
-        plan_action(found_action, repository, smart_proxy, *args)
+        if return_action_output
+          sequence do
+            if pass_only_args
+              input[:output] = plan_action(found_action, *args).output
+              plan_self(input)
+            else
+              input[:output] = plan_action(found_action, repository, smart_proxy, *args).output
+              plan_self(input)
```

Comment:
I don't think you need to pass 'input' here, 'plan_self' should be fine

Reply:
oh wait nevermind, i see what you're doing.   The storing of output on the input is kind confusing, can you just do something like:
```
action_output = plan_action(found_action, *args).output
plan_self(:output => action_output)
```

I think that is clearer as to what is going on